### PR TITLE
skuRecommendationPage & azdata change

### PR DESCRIPTION
**Pull request by aaomidi**
This PR fixes #

### CODE REVIEW

File: extensions/sql-migration/src/models/migrationWizardPage.ts [hunk 1]
```diff
@@ -6,9 +6,24 @@
...
-	public abstract async registerWizardContent(): Promise<void>;
+	public registerWizardContent(): Promise<void> {
+		return new Promise<void>(async (resolve, reject) => {
+			this.wizardPage.registerContent(async (view) => {
+				try {
+					await this.registerContent(view);
+					resolve();
+				} catch (ex) {
+					reject(ex);
+				} finally {
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/models/migrationWizardPage.ts:19]: Not sure how this finally could ever be reached

File: extensions/sql-migration/src/wizard/skuRecommendationPage.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { MigrationWizardPage } from '../models/migrationWizardPage';
+import { MigrationStateModel, StateChangeEvent } from '../models/stateMachine';
+import { Product } from '../models/product';
+import { CONGRATULATIONS, SKU_RECOMMENDATION_PAGE_TITLE, SKU_RECOMMENDATION_ALL_SUCCESSFUL } from '../models/strings';
+
+export class SKURecommendationPage extends MigrationWizardPage {
+	// For future reference: DO NOT EXPOSE WIZARD DIRECTLY THROUGH HERE.
```
Comment (Charles-Gagnon) [inline extensions/sql-migration/src/wizard/skuRecommendationPage.ts:13]: Why?

Comment (aaomidi) [reply extensions/sql-migration/src/wizard/skuRecommendationPage.ts:13]: just to maintain cleaner code, if something needs to happen with the wizard object itself it should be abstracted out to the state model (its just for my own reference in the future) a lot of this code is temporary :D

File: extensions/sql-migration/src/models/migrationWizardPage.ts (hunk 1 above)
Comment (aaomidi) [reply extensions/sql-migration/src/models/migrationWizardPage.ts:19]: yeah it probably won't be, relic of another time I was putting stuff in the catch

File: extensions/sql-migration/src/wizard/skuRecommendationPage.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/sql-migration/src/wizard/skuRecommendationPage.ts:13]: If you plan on taking this out that's fine - but if this is going to stay in then I'd suggest adding a bit of context here

**Pull request merged**